qualtran/bloqs/gf_arithmetic/gf2_multiplication.py

```diff
@@ -364,7 +371,7 @@ def build_call_graph(
         return {}
 
     def __hash__(self):
-        return hash((self.const.additive_order, self.galois_field.irreducible_poly))
+        return hash((self.const.tobytes(), self.m_x))
```

Comment:
The custom hash would go away once we change `const` to be of type `SymbolicInt`.